### DISCUSSION TITLE
Slowing down of progress?

### DISCUSSION THREAD

**Discussion by timsneath** [microsoft/win32metadata#935] [General] [answered]
This project is amazing and will be a huge step forward for Microsoft when complete, but it's not done. The last mile matters: getting to a consistent, well-documented set of metadata will give Microsoft a stable platform for decades to come. The Win32 API is the most durable asset in Windows, as history has shown. 

I've noticed that the pace of commits has dropped, and the issue log has crept up. Please tell me that this is a temporary interruption in service, rather than a shifting of focus? The issue log shows that this work is not complete, and I hope that Microsoft doesn't think that so long as Rust and C# are covered, other languages don't matter!

**Comment by sotteson1** [comment, marked as answer]
Thanks for the question, Tim. We look at this project as something that has a long tail, because the Win32 surface is so huge. There will always be improvements that can be made such as better constants scraping, fixing API definitions that are missing good SAL annotations, etc. We think the "big rocks" are finished, but we definitely want to address issues that are blocking language projections (and not just Rust and C#). Do you have some that are blocking you?

Those of us working on this project have other responsibilities at Microsoft, so we can't spend 100% of our time here, but we do want to make time for blocking issues. We want to fix non-blocking issues as well as our time permits, but probably not as fast as people would like.

**Comment by timsneath** [comment]
You all are amazing: I hope you don't see this as a whine :) Rather, I want to make sure that you hear loud and clear that this is used. I know that there's a tendency at large companies (mine included) for folk to get pulled off projects when they reach a certain point of near-done, and then the project gradually atrophies. I really hope that doesn't happen here, because we are relying on it here at Google. 

In terms of issues, I can obviously selfishly point you at https://github.com/microsoft/win32metadata/issues/created_by/timsneath for a list of things I'd like you to work on :) These are all things that will improve the quality of the Dart projection and/or unblock painful workarounds we have to do on our side.
